File: Service/LivestreamApp.Server/Shutdown/IShutdownService.cs
namespace LivestreamApp.Server.Shutdown
{
    /// <summary>
    ///     Exposes functionality to shutdown or restart the server
    /// </summary>
    public interface IShutdownService
    {
        /// <summary>
        ///     Shuts down the server immediately
        /// </summary>
        void ShutdownServer();

        /// <summary>
        ///     Restarts the server immediately
        /// </summary>
        void RestartServer();
    }
}

File: Service/src/lib/shutdown/i-shutdown-service.ts
/**
 * A contract for shutting down the system
 */
export interface IShutdownService {

    /**
     * Shuts down the system immediately
     */
    shutdown(): void;

    /**
     * Schedules the server shutdown
     *
     * @param time The date time the server shall shut down automatically
     */
    scheduleShutdown(time: Date): void;

    /**
     * Unschedules any scheduled shutdown
     */
    unscheduleShutdown(): void;
}
